presto-main/src/test/java/io/prestosql/server/TestHttpRequestSessionContext.java

```diff
@@ -95,6 +95,32 @@ public void testSessionContext()
         assertEquals(context.getIdentity().getExtraCredentials(), ImmutableMap.of("test.token.foo", "bar", "test.token.abc", "xyz"));
     }
 
+    @Test
+    public void testExtractedUser()
+    {
+        MockHttpServletRequest servletRequest = new MockHttpServletRequest(
+                ImmutableListMultimap.of(PRESTO_USER, "testUser"),
+                "testRemote");
+        HttpRequestSessionContext context = new HttpRequestSessionContext(WARN, servletRequest);
+        assertEquals(context.getIdentity(), Identity.ofUser("testUser"));
+
+        servletRequest = new MockHttpServletRequest(ImmutableListMultimap.of(), "testRemote");
+        servletRequest.setAttribute(PRESTO_USER, "extractedUser");
+        context = new HttpRequestSessionContext(WARN, servletRequest);
+        assertEquals(context.getIdentity(), Identity.ofUser("extractedUser"));
+
+        servletRequest = new MockHttpServletRequest(
+                ImmutableListMultimap.of(PRESTO_USER, "testUser"),
+                "testRemote");
+        servletRequest.setAttribute(PRESTO_USER, "extractedUser");
+        context = new HttpRequestSessionContext(WARN, servletRequest);
+        assertEquals(context.getIdentity(), Identity.ofUser("testUser"));
+
+        assertThatThrownBy(() -> new HttpRequestSessionContext(WARN, new MockHttpServletRequest(ImmutableListMultimap.of(), "testRemote")))
+                .isInstanceOf(WebApplicationException.class)
```

Comment:
```java
.isInstanceOfSatisfying(WebApplicationException.class, e ->
        assertEquals(e.getResponse().getStatus(), 400));
```
Or
```java
.asInstanceOf(WebApplicationException.class)
.satisfies(e -> assertEquals(e.getResponse().getStatus(), 400));
```
Or
```java
.asInstanceOf(WebApplicationException.class)
.extracting(e -> e.getResponse().getStatus())
.isEqualTo(400);
```

Reply:
`asInstanceOf` doesn't seem to work that way.  It needs factory of some kind.  I'm going to leave this for now.